server/server.vala

```diff
@@ -448,8 +448,9 @@ namespace Frida.Server {
 					path.scanf ("/re/frida/AgentSession/%u", out session_id);
 					if (member == "Close") {
 						if (type != DBusMessageType.METHOD_CALL) {
+                            var client = this;
 							Idle.add (() => {
```

Comment:
you mean something like this? : 

```
public void schedule_idle (owned SourceFunc function) {
	Idle.add((owned) function);
}
```

and then:

```
	schedule_idle (() => {
		sessions.remove (session_id);
		return false;
	});
```

but that alone isn't enough to keep a reference to `this`... it would be great do do something inside `schedule_idle` to take care of that